### DISCUSSION TITLE
rtabmap don't run on raspberrypi 4b 4GB and ubuntu 22.04 beacuse of graphic driver and opengl version

### DISCUSSION THREAD

**Discussion by damavand1** [introlab/rtabmap#1059] [General]
hi  all

I installed rtabmap but rtabmap don't run on raspberrypi 4b 4GB on ubuntu 22.04 beacuse of graphic driver and opengl version.
this problem happened only if I connect monitor to micro hdmi port, in other hand if i install xrdp on my ubuntu and connect to it using remote desktop and enter rtabmap in terminal, rtabmap will run 

my another problem is rtabmap play button for start to record mapping is disabled, my camera is xbox 360 Kinect and i installed libfreenect before install rtabmap, is there any confing needed to detect my camera ?

**Comment by matlabbe** [comment]
If play button is disabled, you have to click on "New database" button before.

rtabmap visualization on RPI is pretty slow. I am not sure what is your end goal, but I would install ROS packages directly, launch rtabmap node on RPI and launch visualization / rviz on another computer having a graphic card.

**Comment by damavand1** [comment]
this is the error i got on RPI 4 b 4GB

![WIN_20230614_16_04_59_Pro](https://github.com/introlab/rtabmap/assets/104978921/9a580b1b-ec56-4059-9d96-1ecc7b698d48)

**Comment by damavand1** [comment]
I got OpenGL error on raspberry Pi 4
but i have not any error when i connect to my RPI 4 using remote desktop and every thing go well 
but when i connect my RPi to monitor directly using HTMI i got openGL error, how can i solve this problem ?

**Reply by matlabbe** [reply]
I get the same OpenGL error on ubuntu mate 20.04, but the app is still working, so it looks more a warning than an error.
